### ISSUE TITLE
Redisearch compile issues on MacOS Monterrey 12.2.1

### ISSUE THREAD

**Issue by somnathchoudhuri** [RediSearch/RediSearch#2688]
[ 30%] Building C object CMakeFiles/rscore.dir/src/score_explain.o
[ 30%] Building C object CMakeFiles/rscore.dir/src/rwlock.o
/Users/somnathchoudhuri/software/redisearch/RediSearch/src/hybrid_reader.c:154:9: error: implicit declaration of function 'isnanf' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    if (isnanf(dist)) {
        ^
/Users/somnathchoudhuri/software/redisearch/RediSearch/src/hybrid_reader.c:385:17: warning: incompatible function pointer types assigning to 't_docId (*)(void *)' (aka 'unsigned long long (*)(void *)') from 'size_t (void *)' (aka 'unsigned long (void *)') [-Wincompatible-function-pointer-types]
  ri->LastDocId = HR_LastDocId;
                ^ ~~~~~~~~~~~~
1 warning and 1 error generated.
make[3]: *** [CMakeFiles/rscore.dir/build.make:1112: CMakeFiles/rscore.dir/src/hybrid_reader.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/Makefile2:401: CMakeFiles/rscore.dir/all] Error 2
make[1]: *** [Makefile:146: all] Error 2
make: *** [Makefile:367: /Users/somnathchoudhuri/software/redisearch/RediSearch/bin/macos-arm64v8-release/search/redisearch.so] Error 2

**Comment by somnathchoudhuri** [comment]
Issue is with -Werror,-Wimplicit-function-declaration and gcc/clang version installed with xcode command line tools on MacOS 12.2.1 (Monterey). Request REDIS to please issue a patch for this.

**Comment by somnathchoudhuri** [comment]
Tried with 2.2.10

somnathchoudhuri@Somnaths-MBP RediSearch-2.2.10 % make setup
Makefile:59: deps/readies/mk/main: No such file or directory
Makefile:294: /defs: No such file or directory
Makefile:324: /rules: No such file or directory
./sbin/pack.sh: line 9: /Users/somnathchoudhuri/software/RediSearch-2.2.10/deps/readies/shibumi/defs: No such file or directory
./sbin/pack.sh: line 53: /Users/somnathchoudhuri/software/RediSearch-2.2.10/deps/readies/bin/platform: No such file or directory
./sbin/pack.sh: line 56: /Users/somnathchoudhuri/software/RediSearch-2.2.10/deps/readies/bin/platform: No such file or directory
./sbin/pack.sh: line 59: /Users/somnathchoudhuri/software/RediSearch-2.2.10/deps/readies/bin/platform: No such file or directory
/Users/somnathchoudhuri/software/RediSearch-2.2.10/sbin/getver: line 7: /Users/somnathchoudhuri/software/RediSearch-2.2.10/deps/readies/shibumi/defs: No such file or directory
mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix 
/Users/somnathchoudhuri/software/RediSearch-2.2.10/sbin/getver: line 22: $getver_c: ambiguous redirect
clang: error: no input files
/Users/somnathchoudhuri/software/RediSearch-2.2.10/sbin/getver: line 35: /var/folders/f1/q0_dhgz17nq8mmrsgff82x_w0000gn/T//getver.S5ULMtI: Permission denied
/Users/somnathchoudhuri/software/RediSearch-2.2.10/sbin/getver: line 7: /Users/somnathchoudhuri/software/RediSearch-2.2.10/deps/readies/shibumi/defs: No such file or directory
mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix 
/Users/somnathchoudhuri/software/RediSearch-2.2.10/sbin/getver: line 11: $getver_c: ambiguous redirect
clang: error: no input files
/Users/somnathchoudhuri/software/RediSearch-2.2.10/sbin/getver: line 35: /var/folders/f1/q0_dhgz17nq8mmrsgff82x_w0000gn/T//getver.7vyIuFa: Permission denied
fatal: not a git repository (or any of the parent directories): .git
make: *** No rule to make target `/rules'.  Stop.
somnathchoudhuri@Somnaths-MBP RediSearch-2.2.10 %

**Comment by gkorland** [comment]
The you fetch all the git sub-modules?

**Comment by somnathchoudhuri** [comment]
Yes - the same code is compiling A-OK and working on Big Sur on a 2019 Macbook Pro, without any issues. The issue is with C99 and a higher version of gcc.

**Comment by somnathchoudhuri** [comment]
Hi Redisearch Team - any ETA for a fix?
